src/backend/embedding/chunkfile_Weaviateprocess/chunkfile_uploadWeaviate.py

```diff
@@ -0,0 +1,115 @@
+            raise e
+
+    def insert_data(self, customer_guid: str, file_path: str):
+
+        try:
+            class_names=self.weaviate.generate_weaviate_class_name(customer_guid)
+
+            with open(file_path, "r", encoding="utf-8") as file:
+                data = json.load(file)
+            if not isinstance(data, list):
+                raise ValueError("Invalid JSON format: Expected a list of objects.")
+
+            for index, entry in enumerate(data):
+                if "text" not in entry or "metadata" not in entry:
+                    raise ValueError(f"Missing required fields in entry {index}")
+
+                text = entry["text"].strip()
+                metadata = entry["metadata"]
+
+                if not metadata.get("filename"):
+                    raise ValueError(f"Missing filename in metadata for entry {index}")
+
+                embedding = self.model.encode(text).tolist()
+
+                chunk_data = {
+                    "text": text,
+                    "customer_guid": customer_guid,
+                    "filename": metadata["filename"],
+                    "chunk_number": metadata["chunk_number"],
+                    "page_numbers": metadata["page_numbers"]
+                }
+
+                self.client.data_object.create(chunk_data, class_name=class_names, vector=embedding)
+
+            logger.info(f"Data inserted for {class_names} successfully with external embeddings!")
+        except Exception as e:
+            logger.error(f"Unexpected error inserting data for {customer_guid}: {e}")
+            raise e
+
+    def search_query(self, customer_guid, question, alpha=0.5):
+        try:
+            # Get the query vector for the question
+            query_vector = self.model.encode(question).tolist()
+
+            class_names=self.weaviate.generate_weaviate_class_name(customer_guid)
+
+            # Perform the query with the provided vector
+            result = self.client.query.get(
+               class_names, ["text", "chunk_number", "page_numbers", "filename", "customer_guid"]
+            ).with_hybrid(query=question, alpha=alpha, vector=query_vector).do()
+
+            if not result or "data" not in result or "Get" not in result["data"]:
+                raise ValueError(f"Unexpected search result format: {result}")
+
+            if class_names not in result["data"]["Get"]:
+                raise ValueError(f"No results found for customer: {class_names}")
+
+            for obj in result["data"]["Get"][class_names]:
+                if obj["customer_guid"] != customer_guid:
+                    raise ValueError("Internal server error: Customer GUID mismatch detected!")
+
+            logger.info(f"Search query successful for {customer_guid} with query '{question}'")
+            return result
+
+        except Exception as e:
+            logger.error(f"Unexpected error in search query for {customer_guid}: {e}")
+            raise e
+
+class SearchProcessor:
```

Comment:
We dont need this class SearchProcessor

We can move this functionality under __name__ = main section, if it is used for testing